### PR TITLE
Redis cluster redis classes

### PR DESCRIPTION
**Pull request by chillaq**
# .NET SDK

## Extra Notes

### CODE REVIEW

File: tests/Splitio-tests/Integration Tests/RedisClientTests.cs [hunk 1]
```diff
@@ -71,6 +72,18 @@ public void GetTreatment_WhenFeatureExists_ReturnsOn()
             //Assert
             Assert.IsNotNull(result);
             Assert.AreEqual("on", result);
+
+            var client2 = new RedisClient(GetRedisClusterConfigurationOptions(),
+                API_KEY);
```
Comment (sanzmauro) [inline tests/Splitio-tests/Integration Tests/RedisClientTests.cs:77]:
```suggestion
            var client2 = new RedisClient(GetRedisClusterConfigurationOptions(), API_KEY);
```

File: tests/Splitio-tests/Integration Tests/RedisClientTests.cs [hunk 2]
```diff
@@ -87,6 +100,16 @@ public void GetTreatment_WhenFeatureExists_ReturnsOff()
             //Assert
             Assert.IsNotNull(result);
             Assert.AreEqual("off", result);
+
+            var client2 = new RedisClient(GetRedisClusterConfigurationOptions(),
+                API_KEY);
```
Comment (sanzmauro) [inline tests/Splitio-tests/Integration Tests/RedisClientTests.cs:105]:
```suggestion
            var client2 = new RedisClient(GetRedisClusterConfigurationOptions(), API_KEY);
```

File: tests/Splitio-tests/Integration Tests/RedisClientTests.cs [hunk 3]
```diff
@@ -102,6 +125,16 @@ public void GetTreatment_WhenFeatureDoenstExist_ReturnsControl()
             //Assert
             Assert.IsNotNull(result);
             Assert.AreEqual("control", result);
+
+            var client2 = new RedisClient(GetRedisClusterConfigurationOptions(),
+                API_KEY);
```
Comment (sanzmauro) [inline tests/Splitio-tests/Integration Tests/RedisClientTests.cs:130]: same here

File: tests/Splitio-tests/Integration Tests/RedisClientTests.cs [hunk 4]
```diff
@@ -209,6 +298,18 @@ public void GetTreatments_WhenClientIsNotReady_ReturnsControl()
             {
                 Assert.AreEqual("control", res.Value);
             }
+
+            var client2 = new RedisClient(GetRedisClusterConfigurationOptions(),
+API_KEY);
```
Comment (sanzmauro) [inline tests/Splitio-tests/Integration Tests/RedisClientTests.cs:303]: same here

File: tests/Splitio.Tests.Common/BaseAsyncClientTests.cs [hunk 5]
```diff
@@ -6,6 +6,7 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Splitio.Redis;
```
Comment (sanzmauro) [inline tests/Splitio.Tests.Common/BaseAsyncClientTests.cs:9]: why do you need this?

**Review by sanzmauro — CHANGES_REQUESTED** [review]
Added some suggestions. Also, are you working to adding config validations in upcoming PRs? or should we add in this one?

File: Splitio.Redis/Services/Cache/Classes/BaseAdapter.cs [hunk 6]
```diff
@@ -47,6 +48,16 @@ protected IServer GetServer()
 
             return conn?.GetServer(_config.HostAndPort);
```
[the remark this answers is not in the document]
Comment (chillaq) [reply Splitio.Redis/Services/Cache/Classes/BaseAdapter.cs:49]: I will remove it and will update the code that use it in RedisAdapterConsumer

File: Splitio.Redis/Services/Domain/RedisConfig.cs [hunk 7]
```diff
@@ -35,6 +40,10 @@ public void FromCacheAdapterConfig(CacheAdapterConfigurationOptions options)
             RedisUserPrefix = options.UserPrefix;
             TlsConfig = options.TlsConfig;
             PoolSize = options.PoolSize ?? 1;
+            ClusterNodes = options.RedisClusterNodes != null ? options.RedisClusterNodes.EndPoints : null;
+            KeyHashTag = options.RedisClusterNodes != null ? options.RedisClusterNodes.KeyHashTag : null;
+            ClusterMode = ClusterNodes != null;
```
[the remark this answers is not in the document]
Comment (chillaq) [reply Splitio.Redis/Services/Domain/RedisConfig.cs:43]: I will need to update all code that uses these properties.

File: tests/Splitio-tests/Integration Tests/RedisClientTests.cs (hunk 3 above)
Comment (chillaq) [reply tests/Splitio-tests/Integration Tests/RedisClientTests.cs:128]: will do

File: tests/Splitio-tests/Integration Tests/RedisClientTests.cs (hunk 4 above)
Comment (chillaq) [reply tests/Splitio-tests/Integration Tests/RedisClientTests.cs:301]: ok

File: tests/Splitio.Tests.Common/BaseAsyncClientTests.cs (hunk 5 above)
Comment (chillaq) [reply tests/Splitio.Tests.Common/BaseAsyncClientTests.cs:9]: will remove

**Review by sanzmauro — APPROVED** [review]

**Pull request merged**